libtenzir/builtins/contexts/lookup_table.cpp

```diff
@@ -567,35 +708,68 @@ struct v1_loader : public context_loader {
                                "context entry in serialized entry list, "
                                "entry must be a record");
       }
-      if (not record->fields() or record->fields()->size() != 2) {
+      if (not record->fields() or record->fields()->size() != 4) {
```

Comment:
Instead of getting the values by index, iterate over `*record->fields()`, and act based on the field's name.